### PR TITLE
MeshRenderer Layout

### LINKED ISSUE #403: Rework MeshRenderer layout
MeshRenderer is pretty large at 248 bytes. By moving the strings/paths behind a 'cold' pointer cuts that down to 56. Make the members looks something like:
```cpp
MeshView mesh;
PbrMaterialView materialView;
TechniqueType techniqueType;
std::unique_ptr<ColdState> cold;
```

### PR DESCRIPTION
**Pull request by McCallisterRomer**
resolves #403

Moving path members in `MeshRenderer` behind a pointer to shrink its memory footprint. The common access pattern just looks at the views, while paths are primarily used by the editor. I also removed some of the #ifdef stuff for simplicity.

Fixed some spacing/separator issues with the built in editor.

### CODE REVIEW

File: include/ncengine/graphics/MeshRenderer.h [hunk 1]
```diff
@@ -3,68 +3,87 @@
...
+{
+    NC_ENABLE_IN_EDITOR(MeshRenderer)
+
+    public:
+        MeshRenderer(Entity entity,
+                     std::string meshUid,
+                     PbrMaterial material,
+                     TechniqueType techniqueType = TechniqueType::PhongAndUi);
+
+        auto GetMeshView() const noexcept -> const MeshView&
```
Comment (jaremieromer) [inline include/ncengine/graphics/MeshRenderer.h:42]: Why expand these blocks?

**Review by jaremieromer — COMMENTED** [review]
Can you make these same changes to ToonRenderer as well?

**Comment by McCallisterRomer** [comment]
> Can you make these same changes to ToonRenderer as well?

Definitely, #410.

File: include/ncengine/graphics/MeshRenderer.h (hunk 1 above)
Comment (McCallisterRomer) [reply include/ncengine/graphics/MeshRenderer.h:42]: Can't say I thought too much about it - I added `noexcept` to them and they felt a little long. I could be talked into changing them back.

**Review by jaremieromer — APPROVED** [review]

**Pull request merged**